### PR TITLE
hal support for NETC

### PR DESCRIPTION
**Pull request by yangbolu1991**
This is hal support patch for NETC.
Thanks.

### CODE REVIEW

**Comment by yangbolu1991** [comment]
Updated to v2. Changes,
- Rebased.

Thanks.

**Comment by yangbolu1991** [comment]
Updated to v3. Changes,
- Rebased.

Thanks.

File: mcux/mcux-sdk/drivers/netc/driver_netc.cmake [hunk 1]
```diff
@@ -4,6 +4,7 @@ message("driver_netc component is included.")
 
 target_sources(${MCUX_SDK_PROJECT_NAME} PRIVATE
     ${CMAKE_CURRENT_LIST_DIR}/fsl_netc_endpoint.c
+    ${CMAKE_CURRENT_LIST_DIR}/fsl_netc_timer.c
```
Comment (dleach02) [inline mcux/mcux-sdk/drivers/netc/driver_netc.cmake:7]: has this change been submitted to the SDK?

Need to update mcux/README patchlist with this change so we don't overwrite in a future update if the change isn't in the SDK yet.

**Comment by yangbolu1991** [comment]
Update to v4. Changes,
- Moved compile definitions per @decsny  suggestion.

Thanks.

File: mcux/mcux-sdk/drivers/netc/driver_netc.cmake (hunk 1 above)
Comment (yangbolu1991) [reply mcux/mcux-sdk/drivers/netc/driver_netc.cmake:7]: @dleach02 I have just made the changes merged into mcu sdk.
Thanks.

**Review by decsny — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Pull request merged**